extensions/control-plane/api/management-api/contract-definition-api/src/main/java/org/eclipse/edc/connector/api/management/contractdefinition/model/ContractDefinitionRequestDto.java

```diff
@@ -39,6 +40,9 @@ public class ContractDefinitionRequestDto {
     @NotNull(message = "criteria cannot be null")
     private List<CriterionDto> criteria = new ArrayList<>();
 
+    @Positive(message = "contractValidityDuration must be positive")
+    private long contractValidityDuration;
```

Comment:
I was wondering if this could be represented in a more structured way, because there could be different levels of duration, someone could need a contract valid for some minutes (so like 600 seconds), but it could be also 1 or more years (1 year is 31536000 seconds), it will make the api not so user friendly, so my proposal could be to introduce a structured type, at least on the `Dto` level, like:
```java
public class DurationDto {
  public int amount;
  public String unit;
}
```

and unit could accept values as "minutes", "hours", "days", "months", "years".
Then we could decide if in the `ContractDefinition` class we want to translate it into seconds or into a java `Duration` object, maybe for serialization the first one would be simpler.

Reply:
Done